Common/Base/TexToGen.cs

```diff
@@ -21,14 +22,23 @@ public static TexGen GetTexGenerator(Texture2D tileTex, Dictionary<Color, int> c
                 colorToLiquid[new Color(255, 255, 0)] = 2;
                 colorToLiquid[new Color(150, 150, 150)] = -2;
             }
+
+
             Color[] tileData = new Color[tileTex.Width * tileTex.Height];
-            tileTex.GetData(0, tileTex.Bounds, tileData, 0, tileTex.Width * tileTex.Height);
             Color[] wallData = (wallTex != null ? new Color[wallTex.Width * wallTex.Height] : null);
-            if (wallData != null) wallTex.GetData(0, wallTex.Bounds, wallData, 0, wallTex.Width * wallTex.Height);
             Color[] liquidData = (liquidTex != null ? new Color[liquidTex.Width * liquidTex.Height] : null);
-            if (liquidData != null) liquidTex.GetData(0, liquidTex.Bounds, liquidData, 0, liquidTex.Width * liquidTex.Height);
             Color[] slopeData = (slopeTex != null ? new Color[slopeTex.Width * slopeTex.Height] : null);
-            if (slopeData != null) slopeTex.GetData(0, slopeTex.Bounds, slopeData, 0, slopeTex.Width * slopeTex.Height);
+            var evt = new ManualResetEvent(false);
+            Main.QueueMainThreadAction(() =>
+            {
+                tileTex.GetData(0, tileTex.Bounds, tileData, 0, tileTex.Width * tileTex.Height);
+                if (wallData != null) wallTex.GetData(0, wallTex.Bounds, wallData, 0, wallTex.Width * wallTex.Height);
+                if (liquidData != null) liquidTex.GetData(0, liquidTex.Bounds, liquidData, 0, liquidTex.Width * liquidTex.Height);
+                if (slopeData != null) slopeTex.GetData(0, slopeTex.Bounds, slopeData, 0, slopeTex.Width * slopeTex.Height);
+                evt.Set();
```

Comment:
Worldgen no longer runs on the main thread, so we have to do this messy thing to extract data from the textures.